sympy/geometry/ellipse.py

```diff
@@ -433,6 +465,45 @@ def equation(self, x='x', y='y'):
         t2 = ((y - self.center.y) / self.vradius)**2
         return t1 + t2 - 1
 
+    def equation_using_slope(self, slope):
+        """
+        Returns the equation of an ellipse using the slope of major axis, along with hradius
+        and vradius.
+
+        Approach to the solution :
+
+        Let the center of the ellipse be at C=(xc,yc)
+        Let the major axis be the line that passes through C with a slope of s;
+        points on that line are given by the zeros of L(x,y) = y−yc − s*(x−xc).
+        Let the minor axis be the line perpendicular to L (and also passing through C);
+        points on that line are given by the zeros of l(x,y) = s*(y−yc) + (x−xc).
+
+        Requiring that the distance between the intersections of E and L be 2M identifies
+        b = M**2*(1 + s**2)
+
+        and similarly, requiring that the intersections between E and l be separated by 2m identifies
+        a = m**2*(1 + s**2)
+
+        So the general equation of the ellipse centered at (xc,yc) whose major axis (with radius of M)
```

Comment:
```
So the general equation of the ellipse centered at (xc,yc) whose major axis (with radius of M)
        is on a line with slope s, and whose minor axis has radius of m, is given by the solutions of:
         The ellipse is then defined by the zeros of
```

```
So the points that are on an ellipse centered at (xc,yc) whose major axis (with radius of M)
        is on a line with slope s, and whose minor axis has radius of m, are given be the zeros of:
```
It would be appropriate to reference this SO page. If you search SymPy for references to wkipedia you will find a suitable example.